### ISSUE TITLE
[Factorio] Cannot get game info

### ISSUE THREAD

**Issue by TheCycoONE** [Heroic-Games-Launcher/HeroicGamesLauncher#3635]
### Describe the bug

The game page for Factorio does not populate, instead the message "Cannot get game info" is displayed. Other games are fine, both from GOG and Epic.

I tried gogdl 1.0.1 but the issue persists.

### Add logs

```text
(20:56:45) INFO:    [Legendary]:        Legendary location: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary
(20:56:45) INFO:    [Gog]:              GOGDL location: /home/stephen/Games/Heroic/gogdl
(20:56:45) INFO:    [Connection]:       Connectivity: check-online
(20:56:45) INFO:    [Connection]:       Pinging external endpoints
(20:56:45) INFO:    [Gog]:              Checking for existing gog manifests
(20:56:45) INFO:    [Backend]:          DRM module staus {
  "oimompecagnajdejgnnjijobebaeigek": {
    "name": "Widevine Content Decryption Module",
    "status": "new",
    "version": "4.10.2710.0"
  }
}
(20:56:45) WARNING: [Backend]:          Failed to register protocol with OS.
(20:56:45) INFO:    [Legendary]:        Running command: LEGENDARY_CONFIG_PATH=/home/stephen/.var/app/com.heroicgameslauncher.hgl/config/heroic/legendaryConfig/legendary /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary --version
(20:56:45) INFO:    [Gog]:              Running command: GOGDL_CONFIG_PATH=/home/stephen/.var/app/com.heroicgameslauncher.hgl/config/heroic/gogdlConfig /home/stephen/Games/Heroic/gogdl --auth-config-path /home/stephen/.var/app/com.heroicgameslauncher.hgl/config/heroic/gog_store/auth.json --version
(20:56:45) INFO:    [Nile]:             Running command: NILE_CONFIG_PATH=/home/stephen/.var/app/com.heroicgameslauncher.hgl/config/heroic/nile_config /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/nile --version
(20:56:45) INFO:    [Backend]:          
System Information:
CPU: 16x AMD Ryzen 7 7800X3D 8-Core Processor
Memory: 32.75 GB (used: 4.77 GB)
GPUs:

OS: Arch Linux 6.7.9-arch1-1 (linux)

The current system is not a Steam Deck
We are running inside a Flatpak container

Software Versions:
  Heroic: 2.13.0 Dorry & Broggy
  Legendary: 0.20.34 Direct Intervention
  gogdl: 1.0.1
  Nile: 1.0.1 Jonathan Joestar

(20:56:45) INFO:    [Frontend]:         Refreshing undefined Library
(20:56:45) WARNING: [Backend]:          refresh not implemented on Sideload Library Manager
(20:56:45) INFO:    [Legendary]:        Refreshing library...
(20:56:45) INFO:    [Legendary]:        Refreshing Epic Games...
(20:56:45) INFO:    [Backend]:          Frontend Ready
(20:56:45) INFO:    [Connection]:       Connectivity: online
(20:56:45) INFO:    [Gog]:              Running command: GOGDL_CONFIG_PATH=/home/stephen/.var/app/com.heroicgameslauncher.hgl/config/heroic/gogdlConfig /home/stephen/Games/Heroic/gogdl --auth-config-path /home/stephen/.var/app/com.heroicgameslauncher.hgl/config/heroic/gog_store/auth.json auth
(20:56:45) INFO:    [Gog]:              Checking if login is valid
(20:56:45) INFO:    [Gog]:              Running command: GOGDL_CONFIG_PATH=/home/stephen/.var/app/com.heroicgameslauncher.hgl/config/heroic/gogdlConfig /home/stephen/Games/Heroic/gogdl --auth-config-path /home/stephen/.var/app/com.heroicgameslauncher.hgl/config/heroic/gog_store/auth.json auth
(20:56:45) INFO:    [Backend]:          Checking for current version changelog
(20:56:46) INFO:    [Backend]:          AreWeAntiCheatYet data downloaded
(20:56:46) INFO:    [Legendary]:        Running command: LEGENDARY_CONFIG_PATH=/home/stephen/.var/app/com.heroicgameslauncher.hgl/config/heroic/legendaryConfig/legendary /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
(20:56:47) INFO:    [Gog]:              Saved username to config file
(20:56:47) INFO:    [Legendary]:        Game list updated, got 94 games & DLCs
(20:56:47) WARNING: [Backend]:          listUpdateableGames not implemented on Sideload Library Manager
(20:56:47) INFO:    [Gog]:              Running command: GOGDL_CONFIG_PATH=/home/stephen/.var/app/com.heroicgameslauncher.hgl/config/heroic/gogdlConfig /home/stephen/Games/Heroic/gogdl --auth-config-path /home/stephen/.var/app/com.heroicgameslauncher.hgl/config/heroic/gog_store/auth.json auth
(20:56:47) INFO:    [Gog]:              Found 0 game(s) to update
(20:56:47) INFO:    [Legendary]:        Checking for game updates: LEGENDARY_CONFIG_PATH=/home/stephen/.var/app/com.heroicgameslauncher.hgl/config/heroic/legendaryConfig/legendary /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
(20:56:48) INFO:    [Winetricks]:       Downloading Winetricks
(20:56:48) INFO:    [DXVKInstaller]:    Updating dxvk-nvapi to: dxvk-nvapi-v0.7.0
(20:56:48) INFO:    [DXVKInstaller]:    Updating vkd3d to: vkd3d-proton-2.12
(20:56:48) INFO:    [Legendary]:        Found 0 games to update
(20:56:50) INFO:    [DownloadManager]:  Downloader: Finished downloading https://github.com/Heroic-Games-Launcher/vkd3d-proton/releases/download/v2.12/vkd3d-proton-2.12.tar.xz
(20:56:50) INFO:    [DXVKInstaller]:    Downloaded vkd3d, extracting...
(20:56:50) INFO:    [Backend]:          Extracting /home/stephen/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/vkd3d/vkd3d-proton-2.12.tar.xz to /home/stephen/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/vkd3d/vkd3d-proton-2.12 using native tar
(20:56:50) INFO:    [DXVKInstaller]:    vkd3d updated!
(20:56:50) INFO:    [DownloadManager]:  Downloader: Finished downloading https://github.com/jp7677/dxvk-nvapi/releases/download/v0.7.0/dxvk-nvapi-v0.7.0.tar.gz
(20:56:50) INFO:    [DXVKInstaller]:    Downloaded dxvk-nvapi, extracting...
(20:56:50) INFO:    [Backend]:          Extracting /home/stephen/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/dxvk-nvapi/dxvk-nvapi-v0.7.0.tar.gz to /home/stephen/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/dxvk-nvapi/dxvk-nvapi-v0.7.0 using native tar
(20:56:50) INFO:    [DXVKInstaller]:    dxvk-nvapi updated!
(20:56:50) INFO:    [WineDownloader]:   Updating wine versions info
(20:56:50) INFO:    [WineDownloader]:   Fetching upstream information...
(20:56:50) INFO:    [Backend]:          Starting the Download Queue
(20:56:51) INFO:    [WineDownloader]:   wine versions updated
(20:57:00) INFO:    [ExtraGameInfo]:    Using cached ExtraGameInfo data for Factorio
(20:57:00) INFO:    [Gog]:              Running command: GOGDL_CONFIG_PATH=/home/stephen/.var/app/com.heroicgameslauncher.hgl/config/heroic/gogdlConfig /home/stephen/Games/Heroic/gogdl --auth-config-path /home/stephen/.var/app/com.heroicgameslauncher.hgl/config/heroic/gog_store/auth.json auth
(20:57:00) INFO:    [ExtraGameInfo]:    Using cached ExtraGameInfo data for Factorio
(20:57:00) INFO:    [Gog]:              Running command: GOGDL_CONFIG_PATH=/home/stephen/.var/app/com.heroicgameslauncher.hgl/config/heroic/gogdlConfig /home/stephen/Games/Heroic/gogdl --auth-config-path /home/stephen/.var/app/com.heroicgameslauncher.hgl/config/heroic/gog_store/auth.json auth
(20:57:00) INFO:    [Gog]:              Running command: GOGDL_CONFIG_PATH=/home/stephen/.var/app/com.heroicgameslauncher.hgl/config/heroic/gogdlConfig /home/stephen/Games/Heroic/gogdl --auth-config-path /home/stephen/.var/app/com.heroicgameslauncher.hgl/config/heroic/gog_store/auth.json auth
(20:57:00) INFO:    [Gog]:              Getting game metadata: GOGDL_CONFIG_PATH=/home/stephen/.var/app/com.heroicgameslauncher.hgl/config/heroic/gogdlConfig /home/stephen/Games/Heroic/gogdl --auth-config-path /home/stephen/.var/app/com.heroicgameslauncher.hgl/config/heroic/gog_store/auth.json info 1238653230 --os linux
(20:57:00) INFO:    [Gog]:              Getting game metadata: GOGDL_CONFIG_PATH=/home/stephen/.var/app/com.heroicgameslauncher.hgl/config/heroic/gogdlConfig /home/stephen/Games/Heroic/gogdl --auth-config-path /home/stephen/.var/app/com.heroicgameslauncher.hgl/config/heroic/gog_store/auth.json info 1238653230 --os linux
(20:57:08) ERROR:   [Gog]:              stdout = false and res.abort = false in getInstallInfo
(20:57:08) ERROR:   [Gog]:              stdout = false and res.abort = false in getInstallInfo
(20:57:08) ERROR:   [Frontend]:         Cannot get game info
(20:57:08) ERROR:   [Frontend]:         Cannot get game info
(20:57:08) ERROR:   [Frontend]:         Cannot get game info
(21:00:15) INFO:    [Legendary]:        Running command: LEGENDARY_CONFIG_PATH=/home/stephen/.var/app/com.heroicgameslauncher.hgl/config/heroic/legendaryConfig/legendary /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary --version
(21:00:15) INFO:    [Gog]:              Running command: GOGDL_CONFIG_PATH=/home/stephen/.var/app/com.heroicgameslauncher.hgl/config/heroic/gogdlConfig /home/stephen/Games/Heroic/gogdl --auth-config-path /home/stephen/.var/app/com.heroicgameslauncher.hgl/config/heroic/gog_store/auth.json --version
(21:00:15) INFO:    [Nile]:             Running command: NILE_CONFIG_PATH=/home/stephen/.var/app/com.heroicgameslauncher.hgl/config/heroic/nile_config /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/nile --version
```

### Steps to reproduce

1. Go to Factorio (GOG) in game library
2. Wait a few seconds or attempt to download
3. See error

### Expected behavior

Game page should show and allow download

### Heroic Version

Latest Stable

### System Information

- OS: Arch Linux

### Additional information

```
com.heroicgameslauncher.hgl Heroic]$ GOGDL_CONFIG_PATH="$XDG_CONFIG_HOME/heroic/gogdlConfig" ./gogdl --auth-config-path "$XDG_CONFIG_HOME/heroic/gog_store/auth.json" --version
1.0.1
```

Debug info:
```
[📦 com.heroicgameslauncher.hgl Heroic]$ GOGDL_CONFIG_PATH="$XDG_CONFIG_HOME/heroic/gogdlConfig" ./gogdl --auth-config-path "$XDG_CONFIG_HOME/heroic/gog_store/auth.json" info 1238653230 --platform linux -d
[MAIN] DEBUG: Namespace(display_version=False, auth_config_path='/home/stephen/.var/app/com.heroicgameslauncher.hgl/config/heroic/gog_store/auth.json', command='info', dlcs=False, dlcs_list=None, dlc_only=False, id='1238653230', platform='linux', build=None, lang=None, branch=None, password=None, force_generation=None, workers_count=16)
[GENERIC DOWNLOAD_MANAGER] INFO: Platform is Linux, redirecting download to Linux Native installer manager
[urllib3.connectionpool] DEBUG: Starting new HTTPS connection (1): content-system.gog.com:443
[urllib3.connectionpool] DEBUG: https://content-system.gog.com:443 "GET /products/1238653230/os/windows/builds?generation=2 HTTP/1.1" 200 5576
[urllib3.connectionpool] DEBUG: Starting new HTTPS connection (1): gog-cdn-fastly.gog.com:443
[urllib3.connectionpool] DEBUG: https://gog-cdn-fastly.gog.com:443 "GET /content-system/v2/meta/68/37/6837d6512c1ab9960c3c326ba8eafbfe HTTP/1.1" 200 614
[LINUX] INFO: Initialized Linux Download Manager
[API] INFO: Getting info from products endpoint for id: 1238653230
[urllib3.connectionpool] DEBUG: Starting new HTTPS connection (1): api.gog.com:443
[urllib3.connectionpool] DEBUG: https://api.gog.com:443 "GET /products/1238653230?expand=downloads,expanded_dlcs HTTP/1.1" 200 6180
[API] DEBUG: https://api.gog.com/products/1238653230?expand=downloads,expanded_dlcs
[urllib3.connectionpool] DEBUG: https://api.gog.com:443 "GET /products/1238653230/downlink/installer/en3installer0 HTTP/1.1" 200 1612
[urllib3.connectionpool] DEBUG: https://content-system.gog.com:443 "GET /products/1238653230/users/46987675692784400/namespaces/website/download?path=%2Fsecure%2Flinux_offlines%2F1238653230%2F1238653230%2F57264680172234622%2F2749%2Ffactorio_1_1_104_71197.sh&token=eyJhbGciOiJIUzI1NiIsImV4cCI6IjE3MTA3MjM5ODkifQ.eyJjb250ZW50Ijoie1widXNlcl9pZFwiOjQ2OTg3Njc1NjkyNzg0NDAwLFwicHJvZHVjdF9pZFwiOjEyMzg2NTMyMzAsXCJwYXRoXCI6XCJcXFwvc2VjdXJlXFxcL2xpbnV4X29mZmxpbmVzXFxcLzEyMzg2NTMyMzBcXFwvMTIzODY1MzIzMFxcXC81NzI2NDY4MDE3MjIzNDYyMlxcXC8yNzQ5XFxcL2ZhY3RvcmlvXzFfMV8xMDRfNzExOTcuc2hcIn0iLCJleHAiOiIxNzEwNzIzOTg5In0.YiqkuFOVjIppCS4WBqneMPaSQAS3VKyPGnyutUM6FKo&access_token=RsPxqs7tE3nl-Geks_KHiVlv8ARS1u3diL0wKPExyiGu5D71JDfA4dp1RzM4JK0gdHdEf61wmIFDjHJ89VrMq8zMHCZ_9SSNBHwLPGotm5QSHAdhWC2HGl3gC5XflqGdhKaS45UhC2b_zYCfcgazBw HTTP/1.1" 302 960
[urllib3.connectionpool] DEBUG: https://gog-cdn-fastly.gog.com:443 "GET /token=nva=1710723989~dirs=6~token=0094e57d65e6caeb19a8b/secure/linux_offlines/1238653230/1238653230/57264680172234622/2749/factorio_1_1_104_71197.sh HTTP/1.1" 206 524288
[urllib3.connectionpool] DEBUG: https://gog-cdn-fastly.gog.com:443 "GET /token=nva=1710723989~dirs=6~token=0094e57d65e6caeb19a8b/secure/linux_offlines/1238653230/1238653230/57264680172234622/2749/factorio_1_1_104_71197.sh HTTP/1.1" 206 100
[urllib3.connectionpool] DEBUG: https://gog-cdn-fastly.gog.com:443 "GET /token=nva=1710723989~dirs=6~token=0094e57d65e6caeb19a8b/secure/linux_offlines/1238653230/1238653230/57264680172234622/2749/factorio_1_1_104_71197.sh HTTP/1.1" 206 1975650
[urllib3.connectionpool] DEBUG: Starting new HTTPS connection (1): embed.gog.com:443
[urllib3.connectionpool] DEBUG: https://embed.gog.com:443 "GET /user/data/games HTTP/1.1" 200 5004
Traceback (most recent call last):
  File "gogdl/cli.py", line 67, in <module>
  File "gogdl/cli.py", line 62, in main
  File "gogdl/dl/managers/manager.py", line 58, in calculate_download_size
  File "gogdl/dl/managers/linux.py", line 158, in get_download_size
  File "gogdl/dl/managers/linux.py", line 135, in calculate_download_sizes
AttributeError: 'InstallerHandler' object has no attribute 'central_directory'
[16] Failed to execute script 'cli' due to unhandled exception!
```

**Comment by SteveMunday** [comment]
Confirming i can reproduce on my Steam Deck OLED, this was very frustrating.  Adding my system info.

As a workaround I just opened GOG store in the webview and downloaded the installer manually.  Since its native linux it doesn't need a wine prefix and runs fine.

```
CPU: 8x AMD Custom APU 0932
Memory: 15.53 GB (used: 5.67 GB)
GPUs:
  GPU 0:
    Name: Advanced Micro Devices, Inc. [AMD/ATI] Sephiroth [AMD Custom GPU 0405]
    IDs: D=1435 V=1002 SD=0123 SV=1002
    Driver: amdgpu
OS: SteamOS 3.5.17 holo (linux)

The current system is a Steam Deck (model: OLED) in desktop mode
We are running inside a Flatpak container

Software Versions:
  Heroic: 2.13.0 Dorry & Broggy
  Legendary: 0.20.34 Direct Intervention
  gogdl: 1.0.0
  Nile: 1.0.1 Jonathan Joestar
```

**Comment by imLinguin** [comment]
I just pushed new commit on gogdl repo that should resolve the issue. I don't own the game, so I'm unable to verify this myself  
commit: https://github.com/Heroic-Games-Launcher/heroic-gogdl/commit/c0b996083cb27abebd3811904392b8700071a27a
actions run available here: https://github.com/Heroic-Games-Launcher/heroic-gogdl/actions/runs/8316960653

**Comment by SteveMunday** [comment]
I just installed the new `gogdl` from this build and can confirm that it works for me now, thanks!

**Comment by TheCycoONE** [comment]
Works for me as well, excellent!

**Comment by marcinkoza0922** [comment]
I get this exact same error on Song of Syx and Rimworld as well.

**Comment by benjamin4453** [comment]
I built the gogdl executable following the readme: https://github.com/Heroic-Games-Launcher/heroic-gogdl?tab=readme-ov-file#building-pyinstaller-executable (the actions build has expired).

How can I test it via the Heroic flatpak version?

**Comment by benjamin4453** [comment]
I tested this build and it worked: https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/actions/runs/10305311043?pr=3663

So the issue should be fixed in the next release.

**Comment by arielj** [comment]
closing since the fix for this is included in 2.15.0
